src/codecs/ico/encoder.rs

```diff
@@ -19,6 +20,72 @@ pub struct IcoEncoder<W: Write> {
     w: W,
 }
 
+/// An ICO image entry
+pub struct IcoFrame<'a> {
+    // Pre-encoded PNG or BMP
+    encoded_image: Cow<'a, [u8]>,
+    // Stored as `0 => 256, n => n`
+    width: u8,
+    // Stored as `0 => 256, n => n`
+    height: u8,
+    color_type: ColorType,
+}
+
+impl<'a> IcoFrame<'a> {
+    /// Construct a new `IcoFrame` using a pre-encoded PNG or BMP
+    ///
+    /// The `width` and `height` must be between 1 and 256 (inclusive).
+    pub fn new(
```

Comment:
I'm thinking about the naming and this probably isn't misuse resistant enough :thinking:

Reply:
It's either `new` or a variant of `with_`. So what about `with_encoded`?